e3sm_to_cmip/cmor_handlers/handler.py

```diff
@@ -154,95 +177,544 @@ def to_dict(self) -> Dict[str, Any]:
+                [
+                    v2.LR.historical_0101.eam.h0.1850-01.nc,
+                    v2.LR.historical_0101.eam.h0.1850-02.nc,
+                    ...
+                ]
+            the value at index 0 is v2.LR.historical_0101.eam.h0.1850-01.nc.
+
+        time_dim : str | None
+            Whether or not the output CMIP variable has a time dimension.
+
+        Returns
+        -------
+        xr.Dataset
+            The dataset containing all of the rwar variables.
+        """
+        # Sort the input filepath names for each variable to ensure time axis data
+        # is aligned and in order across variables.
+        sorted_v_to_fp: Dict[str, List[str]] = {
+            var: sorted(vars_to_filepaths[var]) for var in vars_to_filepaths
+        }
+
+        all_filepaths = []
+        for filepaths in sorted_v_to_fp.values():
+            filepath = filepaths[index]
+            all_filepaths.append(filepath)
+
+        ds = xr.open_mfdataset(
+            all_filepaths,
+            decode_times=False,
+            combine="nested",
+            data_vars="minimal",
+            coords="minimal",
+            parallel=True,
         )
 
-    def _write_data(
+        # If the output CMIP variable has an alternative time dimension name (e.g.,
+        # "time2") add that to the xr.Dataset by copying the "time" dimension.
+        if time_dim is not None and time_dim != "time":
+            with xr.set_options(keep_attrs=True):
+                ds = ds.rename({"time": time_dim})
+
+        # Convert "lev" and "ilev" units from mb to Pa for downstream operations.
+        if "lev" in ds:
+            ds["lev"] = ds["lev"] / 1000
+        if "ilev" in ds:
+            ds["ilev"] = ds["ilev"] / 1000
+
+        # If the variable has levels for "sdepth", make sure it has bounds
+        # for the "levgrnd" axis using a statically defined list of bound
+        # values.
+        if self.levels is not None and self.levels["name"] == "sdepth":
+            ds["levgrnd_bnds"] = _formulas.LEVGRND_BNDS
+
+        return ds
+
+    def _get_time_bnds_key(self, data_vars: KeysView[Any]) -> str | None:
+        """Get the key for the time bounds.
+
+          * "atm" variables use "time_bnds"
+          * "ldn" variables use "time_bounds"
+
+        Parameters
+        ----------
+        data_vars : KeysView[Any]
+            A list of data_var keys in the xr.Dataset.
+
+        Returns
+        -------
+        str | None
+           The key of the time bounds, or None if it does not exist.
+        """
+        if "time_bnds" in data_vars:
+            return "time_bnds"
+        elif "time_bounds" in data_vars:
+            return "time_bounds"
+
+        return None
+
+    def _reshape_single_time_bnd(self, ds):
+        # TODO: Reshape time bounds if it exists and has a length of 1: https://github.com/E3SM-Project/e3sm_to_cmip/blob/8c818fcb6d51cc0555f62b8058eee539b69a9579/e3sm_to_cmip/lib.py#L674-L678C42
+        pass
```

Comment:
I added this TODO just in case it is needed. I didn't run into any issues where time bounds have a length of 1 and need to be reshaped like the old CDAT code linked in the comment.